src/main/scala/io/getclump/Clump.scala

```diff
@@ -80,33 +81,33 @@ sealed trait Clump[+T] {
    * A utility method for automatically unwrapping the underlying value
    * @throws NoSuchElementException if the underlying value is not defined
    */
-  def apply(): Future[T] = get.map(_.get)
+  def apply()(implicit ec: ExecutionContext): Future[T] = get.map(_.get)
 
   /**
    * Get the result of the clump or provide a fallback value in the case where the result is not defined
    */
-  def getOrElse[B >: T](default: => B): Future[B] = get.map(_.getOrElse(default))
+  def getOrElse[B >: T](default: => B)(implicit ec: ExecutionContext): Future[B] = get.map(_.getOrElse(default))
 
   /**
    * If the underlying value is a list, then this will return Nil instead of None when the result is not defined
    */
-  def list[B >: T](implicit cbf: CanBuildFrom[Nothing, Nothing, B]): Future[B] =
-    get.map(_.getOrElse(cbf().result))
+  def list[B >: T](implicit cbf: CanBuildFrom[Nothing, Nothing, B], ec: ExecutionContext): Future[B] =
+    get.map(_.getOrElse(cbf().result()))
 
   /**
    * Trigger execution of a clump. The result will not be defined if any of the clump sources returned less elements
    * than requested.
    */
-  def get: Future[Option[T]] =
+  def get(implicit ec: ExecutionContext): Future[Option[T]] =
     new ClumpContext()
       .flush(List(this))
       .flatMap { _ =>
         result
       }
 
-  protected[getclump] def upstream: List[Clump[_]]
-  protected[getclump] def downstream: Future[List[Clump[_]]]
-  protected[getclump] def result: Future[Option[T]]
+  protected[getclump] def upstream: List[Clump[_]] = List.empty[Clump[_]]
+  protected[getclump] def downstream(implicit ec: ExecutionContext): Future[Option[Clump[_]]] = Future.successful(None)
+  protected[getclump] def result(implicit ec: ExecutionContext): Future[Option[T]]
```

Comment:
We could avoid having to pass the execution context here by moving all the future composition logic to a separate class. For example: `class ClumpExecution(implicit ec: ExecutionContext) { def apply[T](clump: Clump[T]) = ... }`. The implementation of `apply` would know how to execute any type of Clump by using pattern matching. `ClumpContext` would create an instance of this new class. 

The change would also helps to reduce this source file size.